packages/sheets/src/model/range-theme-model.ts

```diff
@@ -0,0 +1,272 @@
+            rangeThemeStyleRuleMap[key] = value;
+        });
+
+        // deserialize custom range theme style
+        const rangeThemeStyleMap = this._ensureRangeThemeStyleMap(unitId);
+        const rangeThemeStyleMapJson: Record<string, IRangeThemeStyleJSON> = {};
+        rangeThemeStyleMap.forEach((value, key) => {
+            rangeThemeStyleMapJson[key] = value.toJson();
+        });
+
+        return JSON.stringify({
+            rangeThemeStyleRuleMap,
+            rangeThemeStyleMapJson,
+        });
+    }
+
+    fromJSON(json: string) {
+        const { rangeThemeStyleRuleMap, rangeThemeStyleMapJson } = JSON.parse(json);
+        Object.keys(rangeThemeStyleRuleMap).forEach((key) => {
+            this._ensureRangeThemeStyleRuleMap(key).clear();
+            this._ensureRTreeCollection(key).clear();
+            const ruleMap = rangeThemeStyleRuleMap[key];
+            Object.keys(ruleMap).forEach((ruleKey) => {
+                const rule = ruleMap[ruleKey];
+                this.registerRangeThemeRule(rule.themeName, rule.rangeInfo);
+                const rTreeCollection = this._ensureRTreeCollection(key);
+                rTreeCollection.insert({ unitId: key, sheetId: rule.rangeInfo.subUnitId, range: rule.rangeInfo.range, id: ruleKey });
+            });
+        });
+
+        Object.keys(rangeThemeStyleMapJson).forEach((key) => {
+            const styleMap = rangeThemeStyleMapJson[key];
+            const style = new RangeThemeStyle(styleMap.name);
+            style.fromJson(styleMap);
+            this._ensureRangeThemeStyleMap(key).set(style.getName(), style);
+        });
+    }
+
+    deleteUnitId(unitId: string) {
+        this._rangeThemeStyleMap.delete(unitId);
+        this._rangeThemeStyleRuleMap.delete(unitId);
+        this._rTreeCollection.delete(unitId);
+    }
+
+    private _initSnapshot(): void {
+        this.disposeWithMe(this._resourceManagerService.registerPluginResource({
+            toJson: (unitId: string) => {
+                return this.toJson(unitId);
+            },
+            parseJson: (json: string) => {
+                if (!json) {
+                    return {};
+                }
+                try {
+                    return JSON.parse(json);
+                    // eslint-disable-next-line unused-imports/no-unused-vars
+                } catch (error) {
+                    return {};
+                }
+            },
+            businesses: [UniverInstanceType.UNIVER_SHEET],
+            pluginName: SHEET_RANGE_THEME_MODEL_PLUGIN,
+            onLoad: (_unitId, resources) => {
+                this.fromJSON(resources);
+            },
+            onUnLoad: (unitId) => {
+                this.deleteUnitId(unitId);
+            },
+        }));
+    }
+
+    override dispose(): void {
+        super.dispose();
+        this._rangeThemeStyleMap.clear();
+        this._rangeThemeStyleRuleMap.clear();
+        this._rTreeCollection.clear();
```

Comment:
Maybe we should clear `defaultRangeThemeMap` as well?

Reply:
yes